### ISSUE TITLE
In featureinfo.showInfo, fallback on getfeature if getSource().getFeatureById returns null

### ISSUE THREAD

**Issue by jokd** [origo-map/origo#1438]
There are cases where a feature isn't loaded in the source, eg when using loadingstretegy bbox or using the API. Then it is useful to instead make a getfeature call to get the feature.
Adding a few lines to featureinfo.showInfo would probably solve this and add useful functionality to the API.

**Comment by steff-o** [comment]
Maybe this is related to #1390, or at least they could benefit from each other. 

The getFeature function is buggy and redundant code with wfs and ags loader functions. Therefore I think that the loader function of the source should be exposed in someway in order to reuse it for all kinds of purposes, In my opinion the cleanest way would be to make a class for WFS and one for AGS that inherits VectorSource and exposes some methods to read features from the source, ensure that features have been read, and forcefully re-read loaded features. (Feels like i have written this before...) The other way around would of course also work: fix getFeature and make VectorSource use that as a loader, but looking at it in an object oriented way I think it is more intuitive to read features from a layer's source than calling a utility function, especially when it comes to refreshing and ensure loading.

By pure coincidence the PR #1398 includes such implementation of the source for WFS. It does not however implement all of the above mentioned functions.

**Comment by jokd** [comment]
Yeah the getfeature function could do an overhaul and using the source loader makes a lot sense. But no matter how we handle that, this feature could be added to showInfo instantly. 
Rewriting getfeature and source loaders can take some time and will not necessarily change the API.

**Comment by steff-o** [comment]
Agree that this could be implemented without rewriting the entire loader or getFeature first as getfeature would probably remain as a part of the api anyway. Just be aware that getfeature ignores the projection parameters for WFS, which may put some unexpected behavior on the fallback loading.
